### ISSUE TITLE
no-unused-modules: flag to ignore destructured exports if at least one of them is imported elsewhere

### ISSUE THREAD

**Issue by KristjanTammekivi** [import-js/eslint-plugin-import#2206]
Curernt behavior:

```typescript
export const { somethingUsed, somethingNotUsed } = createSomeStuff(); // exported declaration 'somethingNotUsed' not used within other modules
console.log(somethingNotUsed);
```
I would like it if there was a flag in the rule to not error in this case

**Comment by ljharb** [comment]
Why? It’s not used outside the module, so it shouldn’t be exported. You should do the destructuring in one line, and then `export { somethingUsed }` in another.

**Comment by KristjanTammekivi** [comment]
I guess it's a matter of preference, doing the export on a different line doesn't look as elegant to me and I'm OK with exporting one part of the ?destructure? that isn't used as long as the other is.

**Comment by ljharb** [comment]
In that case it sounds like an eslint override comment would be appropriate?

**Comment by ljharb** [comment]
For the record, this isn't purely a style preference - exporting something you don't use is expanding your API surface, which means you have an additional thing that you can't change without it being breaking, and your application as a whole is more brittle. It's a very bad idea to expose anything that you don't absolutely have to.

**Comment by KristjanTammekivi** [comment]
99% of the time I use this plugin in applications, not libraries so the API surface isn't really an issue.
But fair enough, I guess this is something I have to get used to.
